projects/igniteui-angular/src/lib/grids/state.directive.spec.ts

```diff
@@ -389,14 +389,42 @@ describe('IgxGridState - input properties #grid', () => {
rue,"selectable":true,"parent":"Region Information","columnGroup":false,"disableHiding":false,"disablePinning":false,"visibleWhenCollapsed":true},{"pinned":false,"sortable":false,"filterable":true,"editable":false,"sortingIgnoreCase":true,"filteringIgnoreCase":true,"headerClasses":"","headerGroupClasses":"","groupable":false,"movable":false,"hidden":true,"dataType":"string","hasSummary":false,"width":"0px","header":"City Information","resizable":false,"searchable":true,"selectable":true,"parent":"Country Information","columnGroup":true,"disableHiding":false,"disablePinning":false,"collapsible":false,"expanded":true,"visibleWhenCollapsed":true},{"pinned":false,"sortable":false,"filterable":true,"editable":false,"sortingIgnoreCase":true,"filteringIgnoreCase":true,"headerClasses":"","headerGroupClasses":"","groupable":false,"movable":false,"hidden":true,"dataType":"string","hasSummary":false,"field":"City","width":"100px","header":"","resizable":false,"searchable":true,"selectable":true,"parent":"City Information","columnGroup":false,"disableHiding":false,"disablePinning":false},{"pinned":false,"sortable":false,"filterable":true,"editable":false,"sortingIgnoreCase":true,"filteringIgnoreCase":true,"headerClasses":"","headerGroupClasses":"","groupable":false,"movable":false,"hidden":true,"dataType":"string","hasSummary":false,"field":"Address","width":"100px","header":"","resizable":false,"searchable":true,"selectable":true,"parent":"City Information","columnGroup":false,"disableHiding":false,"disablePinning":false},{"pinned":false,"sortable":false,"filterable":true,"editable":false,"sortingIgnoreCase":true,"filteringIgnoreCase":true,"headerClasses":"","headerGroupClasses":"","groupable":false,"movable":false,"hidden":false,"dataType":"string","hasSummary":false,"width":"200px","header":"Contact Information","resizable":false,"searchable":true,"selectable":true,"parent":"Address Information","columnGroup":true,"disableHiding":false,"disablePinning":false,"collapsible":false,"expanded":true,"visibleWhenCollapsed":false},{"pinned":false,"sortable":false,"filterable":true,"editable":false,"sortingIgnoreCase":true,"filteringIgnoreCase":true,"headerClasses":"","headerGroupClasses":"","groupable":false,"movable":false,"hidden":false,"dataType":"string","hasSummary":false,"field":"Phone","width":"100px","header":"","resizable":false,"searchable":true,"selectable":true,"parent":"Contact Information","columnGroup":false,"disableHiding":false,"disablePinning":false,"visibleWhenCollapsed":false},{"pinned":false,"sortable":false,"filterable":true,"editable":false,"sortingIgnoreCase":true,"filteringIgnoreCase":true,"headerClasses":"","headerGroupClasses":"","groupable":false,"movable":false,"hidden":false,"dataType":"string","hasSummary":false,"field":"Fax","width":"100px","header":"","resizable":false,"searchable":true,"selectable":true,"parent":"Contact Information","columnGroup":false,"disableHiding":false,"disablePinning":false,"visibleWhenCollapsed":false}]}`
```

Comment:
Did you consider the test grid component to consists of much lesser columns, so to end up with relatively short state. Just an option for enhacement.

Reply:
Hm, these public properties (all of them) are used for data binding in the component's template. We need them for setting up the state and behavior of the grid and its columns. Removing them gives me an error.